language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -0,0 +1,493 @@
+import org.sonarsource.sonarlint.core.client.api.common.analysis.AnalysisResults;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.Issue;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneAnalysisConfiguration;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneGlobalConfiguration.Builder;
+import org.sonarsource.sonarlint.core.client.api.standalone.StandaloneSonarLintEngine;
+
+public class SonarLintLanguageServer implements LanguageServer, WorkspaceService, TextDocumentService {
+
+  private static final String DISABLE_TELEMETRY = "disableTelemetry";
+  static final String TEST_FILE_PATTERN = "testFilePattern";
+  static final String ANALYZER_PROPERTIES = "analyzerProperties";
+  private static final String SONARLINT_CONFIGURATION_NAMESPACE = "sonarlint";
+  private static final String SONARLINT_SOURCE = SONARLINT_CONFIGURATION_NAMESPACE;
+  private static final String SONARLINT_OPEN_RULE_DESCRIPTION_COMMAND = "SonarLint.OpenRuleDesc";
+  private static final String DISABLE_PROPERTY_KEY = "sonarlint.telemetry.disabled";
+
+  private final LanguageClient client;
+  private final StandaloneSonarLintEngine engine;
+  private final Future<?> backgroundProcess;
+  private final LanguageClientLogOutput logOutput;
+  private final Map<URI, String> languageIdPerFileURI = new HashMap<>();
+  private final SonarLintTelemetry telemetry = new SonarLintTelemetry();
+
+  private Path workspaceDir;
+  private String testFilePattern;
+  private Map<String, String> analyzerProperties;
+  private int ruleServerPort;
+
+  public SonarLintLanguageServer(int port) throws IOException {
+    Socket socket = new Socket("localhost", port);
+
+    Launcher<LanguageClient> launcher = LSPLauncher.createServerLauncher(this,
+      socket.getInputStream(),
+      socket.getOutputStream(),
+      true, new PrintWriter(System.out));
+
+    this.client = launcher.getRemoteProxy();
+    this.logOutput = new LanguageClientLogOutput(client);
+
+    info("Starting SonarLint engine...");
+
+    try {
+      Builder builder = StandaloneGlobalConfiguration.builder()
+        .setLogOutput(logOutput)
+        .addPlugins(getAnalyzers().toArray(new URL[0]));
+
+      this.engine = new StandaloneSonarLintEngineImpl(builder.build());
+    } catch (Exception e) {
+      error("Error starting SonarLint engine", e);
+      throw new IllegalStateException(e);
+    }
+
+    info("SonarLint engine started");
+
+    backgroundProcess = launcher.startListening();
+  }
+
+  private void debug(String message) {
+    client.logMessage(new MessageParams(MessageType.Log, message));
+  }
+
+  private void info(String message) {
+    client.logMessage(new MessageParams(MessageType.Info, message));
+  }
+
+  private void warn(String message) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "warn" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)